File: UI/src/models/ChatbotConversation.ts
import ChatbotConversationDetail from "./ChatbotConversationDetail";
import MLEntity from "./MLEntity";

interface ChatbotConversation extends MLEntity {
    /// Khoá chính
    ConversationID: string,

    /// ID khách hàng
    CustomerID?: string,

    /// Chủ đề
    Topic: string,

    /// Có hữu ích với người dùng không
    IsHelpful: boolean,

    /// Danh sách tin nhắn trong cuộc hội thoại
    ChatbotConversationDetails: ChatbotConversationDetail[]
}

export default ChatbotConversation;

File: API/ML.BO/ChatbotConversation.cs
using API.ML.BOBase;
using System.ComponentModel.DataAnnotations;

namespace API.ML.BO
{
    public class ChatbotConversation : MLEntity
    {
        /// <summary>
        /// Khoá chính
        /// </summary>
        [Key]
        public Guid ConversationID { get; set; }

        /// <summary>
        /// ID khách hàng
        /// </summary>
        public Guid? CustomerID { get; set; }

        /// <summary>
        /// Chủ đề hỏi
        /// </summary>
        public string Topic { get; set; } = string.Empty;

        /// <summary>
        /// Cuộc hội thoại có hữu ích không
        /// </summary>
        public bool IsHelpful { get; set; }

        /// <summary>
        /// Dữ liệu tin nhắn hội thoại
        /// </summary>
        public IEnumerable<ChatbotConversationDetail>? ChatbotConversationDetails { get; set; }

        /// <summary>
        /// Dữ liệu khách hàng tạo đoạn hội thoại
        /// </summary>
        public Customer? Customer { get; set; }
    }
}
